src/Desktop/ArcGISRuntime.Desktop.Samples/Samples/Map/ManageBookmarks/ManageBookmarks.xaml.cs

```diff
@@ -0,0 +1,159 @@
+            BorderAddBookmark.Visibility = System.Windows.Visibility.Hidden;
+        }
+
+        private void OnBookmarkChooserSelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            // Get the selected bookmarks name.
+            var selectedBookmarkName = e.AddedItems[0].ToString();
+
+            // Get the collection of bookmarks in the map.
+            BookmarkCollection myBookmarkCollection = MyMapView.Map.Bookmarks;
+
+            // Loop through each bookmark. 
+            foreach (var myBookmark in myBookmarkCollection)
+            {
+                // Get the bookmarks name.
+                var theBookmarkName = myBookmark.Name;
+
+                // If the selected bookmarks name matches one in the bookmark collection
+                // set that to be the maps viewpoint.
+                if (theBookmarkName == selectedBookmarkName.ToString())
+                {
+                    MyMapView.SetViewpoint(myBookmark.Viewpoint);
+                }
+            }
+        }
+
+        private void ButtonAddBookmark_Click(object sender, System.Windows.RoutedEventArgs e)
+        {
+            // Show the controls to add a bookmark.
+            BorderAddBookmark.Visibility = System.Windows.Visibility.Visible;
+            ButtonAddBookmark.Visibility = System.Windows.Visibility.Hidden;
+            TextBoxBookmarkName.Text = "";
+        }
+
+
+        private void ButtonCanel_Click(object sender, System.Windows.RoutedEventArgs e)
+        {
+            // Hide the controls to add a bookmark.
+            BorderAddBookmark.Visibility = System.Windows.Visibility.Hidden;
+            ButtonAddBookmark.Visibility = System.Windows.Visibility.Visible;
+        }
+
+        private void ButtonAddDone_Click(object sender, System.Windows.RoutedEventArgs e)
+        {
+            // Get the name from the text field.
+            var name = TextBoxBookmarkName.Text;
+
+            // Exit if the name is empty.
+            if (string.IsNullOrEmpty(name))
+                return;
+
+            // Check to see if there is a bookmark with same name.
+            bool doesNameExist = MyMapView.Map.Bookmarks.Any(b => b.Name == name);
+            if (doesNameExist)
+                return;
+
+            // Create a new bookmark.
+            Bookmark myBookmark = new Bookmark();
+            myBookmark.Name = name;
+
+            // Get the current viewpoint from map and assign it to bookmark. 
+            myBookmark.Viewpoint = MyMapView.GetCurrentViewpoint(ViewpointType.BoundingGeometry);
+            
+            // Add the bookmark to bookmark collection of the map.
+            MyMapView.Map.Bookmarks.Add(myBookmark);
+
+            // Add the bookmark name to the list of choices in the combobox.
+            bookmarkChooser.Items.Add(name);
+
+            // Set the newly added bookmark to be the one selected in the combobox.
+            bookmarkChooser.SelectedValue = name;
```

Comment:
If using Bookmark objects instead of names, you can set the SelectedItem to the new bookmark.
````
bookmarkChooser.SelectedItem = myBookmark;
````